extensions/mssql/src/ui/dialogBase.ts

```diff
@@ -147,6 +147,20 @@ export abstract class DialogBase<DialogResult> {
 		return this.createInputBox(ariaLabel, textChangeHandler, value, enabled, 'password', width);
 	}
 
+	protected createInputBoxWithProperties(textChangeHandler: (newValue: string) => Promise<void>, properties: azdata.InputBoxProperties, customValidation?: () => Promise<boolean>): azdata.InputBoxComponent {
+		properties.width = properties.width ?? DefaultInputWidth;
+		properties.inputType = properties.inputType ?? 'text';
+		properties.value = properties.value ?? '';
+		properties.enabled = properties.enabled ?? true;
+		const inputbox: azdata.InputBoxComponent = customValidation ? this.modelView.modelBuilder.inputBox().withProps(properties).withValidation(customValidation).component() : this.modelView.modelBuilder.inputBox().withProps(properties).component();
```

Comment:
nit: It'd read a little better if you split this up into multiple lines and had an `if (customValidation)` that added the withValidation only if there was one specified.